### PR TITLE
Improve docs

### PR DESCRIPTION
**Pull request by ericphanson**
Add some docstrings, a quick tutorial, and a demo of the tables interplay and parallelization via Transducers

### CODE REVIEW

File: docs/src/index.md [hunk 1]
```diff
@@ -1,13 +1,16 @@
...
 
-```@meta
-CurrentModule = KeywordSearch
-```
+KeywordSearch has two main nouns:
 
-```@contents
-```
+* queries, which come in two flavors, [`Query`](@ref), which is used for exact matches, and [`FuzzyQuery`](@ref) which is for fuzzy matches
+* and documents, which are strings wrapped in [`Document`](@ref) objects and can optionally have metadata associated to them.
```
Comment (hannahilea) [inline docs/src/index.md]:
```suggestion
* and documents, which are strings wrapped in [`Document`](@ref) objects and can optionally have metadata associated with them.
```

File: docs/src/named_queries.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+```@repl 2
+using KeywordSearch, Random, DataFrames
+
+queries =  [NamedQuery(FuzzyQuery("dog"), "find dog"),
+            NamedQuery(FuzzyQuery("cat"), "find cat"),
+            NamedQuery(Query("koala") | FuzzyQuery("Opossum"), "find marsupial")]
+
+words = ["dg", "cat", "koala", "opposum"]
+corpus1 = Corpus([Document(randstring(rand(1:10))*rand(words)*randstring(rand(1:10)), (; doc_index = j)) for j = 1:10], (;name = "docs"))
```
Comment (hannahilea) [inline docs/src/named_queries.md]: YASguide formatting?

File: docs/src/tutorial.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
+# Quick tutorial
+
+In this example, we will use some sample text modified from the public domain text *Aristotle's History of Animals* <http://www.gutenberg.org/files/59058/59058-0.txt>.
```
Comment (hannahilea) [inline docs/src/tutorial.md:3]: :) Someone is having fun!

File: docs/src/named_queries.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+filter!(!isnothing, matches);
+DataFrame(matches)
+```
+
+We can also make use of [Transducers.jl](https://github.com/JuliaFolds/Transducers.jl) to easily
+multithread or parallelize across cores via `tcollect` or `dcollect`:
+
+```@repl 2
+using Transducers
+matches = tcollect(Iterators.product(queries, corpuses) |> MapSplat(match) |> Filter(!isnothing));
```
Comment (hannahilea) [inline docs/src/named_queries.md]: Shhh, don't let Alex see this...

**Review by hannahilea — APPROVED** [review]
Looks good, barring formatting comments!

File: docs/src/named_queries.md (hunk 4 above)
Comment (ericphanson) [reply docs/src/named_queries.md]: yeah.. Transducers code looks weird to me without it but no pipes allowed in YAS so I'll format it out 👍

File: docs/src/named_queries.md (hunk 2 above)
Comment (ericphanson) [reply docs/src/named_queries.md]: good point! pushed

**Pull request merged**